pkg/action/action.go

```diff
@@ -0,0 +1,62 @@
+package action
+
+import (
+	v1api "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/coreos/container-linux-update-operator/pkg/constants"
+)
+
+const (
+	// ActionsUnknown is the zero value for an ActionsResult.
+	// It should not be returned by ActionsCompleted. This value likely
+	// indicates a bug.
+	ActionsUnknown ActionsResult = iota
+	// ActionsPending is returned when there are still actions which haven't
+	// set their configured annotation to true
+	ActionsPending
+	// ActionsSucceeded is returned when all actions have set their configured
+	// annotation to true
+	ActionsSucceeded
+)
+
+// ActionsResult is the result of a given set of actions
+type ActionsResult int
+
+// Action is a pre- or post-reboot action to perform.
+type Action struct {
+	annotation string
+}
+
+// New creates a new action with the given annotation
+func New(annotation string) *Action {
+	return &Action{
+		annotation: annotation,
+	}
+}
+
+// ActionsCompleted inspects a nodes annotations to determine if a set of
+// actions has succeeded, failed, or is pending completion
+// the annotations have three possible states
+//   true - the action has completed successfully
+//   anything else - the action has not yet completed
+// if all of the annotations exist and are true, return ActionsSucceeded
+// if any action does not exist or is not set to true, return ActionsPending
+func ActionsCompleted(node v1api.Node, actions []*Action) ActionsResult {
```

Comment:
In offline conversation we talked about removing this package, removing actions, and just having something kinda like `HasAnnotations(node, annotations []Annotations) bool`